embedded-iconoir/src/icon.rs

```diff
@@ -74,37 +70,39 @@ impl<C: PixelColor, Ico: IconoirInternal> Icon<C, Ico> {
     }
 }
 
+
 pub trait IconoirNewIcon<C: PixelColor>: Sized
 where
-    Self: IconoirInternal,
+    Self: sealed::IconoirInternal,
 {
     fn new(color: C) -> Icon<C, Self>;
 }
 
 impl<C: PixelColor, T> IconoirNewIcon<C> for T
 where
-    T: IconoirInternal,
+    T: sealed::IconoirInternal,
 {
     fn new(color: C) -> Icon<C, Self> {
         Icon {
             color,
-            icon: Self::_new_internal(),
+            _icon: Self::INSTANCE,
         }
     }
 }
 
-pub trait IconoirInternal: Sized {
-    const SIZE: u32;
-    const BIT_COUNT: usize = { Self::SIZE as usize * Self::SIZE as usize };
-    const BYTE_COUNT: usize = { Self::BIT_COUNT / 8 + if Self::BIT_COUNT % 8 > 0 { 1 } else { 0 } };
-    const DATA: &'static [u8];
-
-    fn _new_internal() -> Self;
+pub(crate) mod sealed {
+    pub trait IconoirInternal: Sized {
+        const SIZE: u32;
+        const BIT_COUNT: usize = { Self::SIZE as usize * Self::SIZE as usize };
+        const BYTE_COUNT: usize = { Self::BIT_COUNT / 8 + if Self::BIT_COUNT % 8 > 0 { 1 } else { 0 } };
```

Comment:
Also misc question, is there ever a case where `BYTE_COUNT != DATA.len()`? Might be a fun static assert to have if you aren't sure :D

Reply:
The only situation in which this can occur is if the generator somehow manages to screw up generating the `.bits` files. Which honestly can't hurt to double-check.

Reply:
Ah, I guess you're doing that already on line 25, though you are asserting `>=` instead of `==`.